cmds/webboot/webboot.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	ui "github.com/gizak/termui/v3"
+	"github.com/u-root/webboot/pkg/menu"
+)
+
+const (
+	downloadByLinkLabel = "Download by link"
+	tcURL               = "http://tinycorelinux.net/10.x/x86_64/release/TinyCorePure64-10.1.iso"
+	wbtcpURL            = "https://github.com/u-root/webboot-distro/raw/master/iso/tinycore/10.x/x86_64/release/TinyCorePure64.iso"
+)
+
+// bookmark record the list of iso webboot allow user to download
+var bookmark = []*BookMarkISO{
+	&BookMarkISO{
+		isDefault: false,
+		url:       tcURL,
+		label:     "Download Tinycore v10.1",
+		isoName:   "TinyCorePure64-10.1.iso",
+	},
+	&BookMarkISO{
+		isDefault: false,
+		url:       wbtcpURL,
+		label:     "Webboot Tinycorepure",
+		isoName:   "TinyCorePure64.iso",
+	},
+}
+
+// BookMarkISO contains information of the iso user want to download
+// expected to implement Entry interface
+type BookMarkISO struct {
+	isDefault bool
+	url       string
+	label     string
+	isoName   string
+}
+
+// Label is the string this iso displays in the menu page.
+func (b *BookMarkISO) Label() string {
+	return b.label
+}
+
+// IsDefault is for mark whether this iso is a default choice.
+func (b *BookMarkISO) IsDefault() bool {
```

Comment:
I feel like the isDefault functions are cluttering the file a little - would there be a downside to removing isDefault from the entry and just not having a default?

Reply:
I think it is  only about how user feel. I am not sure will they want a default. I think if the user no nothing about the OS, will it be better to give them a default option? or maybe I can make the first entry in the  menu always be a default? Then we can define the default by changing the order of Entry. what do you think?

Reply:
I think either not having a default or having the first entry be the default is fine.